app/unit-tests/src/org/commcare/android/tests/DemoUserRestoreTest.java

```diff
@@ -93,26 +108,37 @@ public void demoUserRestoreAndUpdateTest() {
         EntityListAdapter adapter = CaseLoadUtils.loadList(entitySelectActivity);
         assertEquals(2, adapter.getCount());
 
+        updatingAppShouldUpdateRecords(sandbox);
+    }
+
+    private void updatingAppShouldUpdateRecords(AndroidSandbox sandbox) {
         // update the app to a version with a new demo user restore
         String profileRef = UpdateUtils.buildResourceRef(REF_BASE_DIR,
                 "update_user_restore", "profile.ccpr");
         UpdateUtils.installUpdate(profileRef,
                 AppInstallStatus.UpdateStaged,
                 AppInstallStatus.Installed);
 
+        Robolectric.flushBackgroundThreadScheduler();
```

Comment:
Is there a reason for these double calls? Might be worth either commenting or wrapping in a method which articulates why we're doing something potentially redundant-looking.

Reply:
And actually, putting the double-call in another method has a second advantage of making it easier to apply the change across all contexts (it's done here at least twice) if the steps for a "Full" flush need to get updated later.

Reply:
Restored to[ singular calls](https://github.com/dimagi/commcare-android/pull/2537/commits/ba3eb0d83ea6cfd64bfa2da2ac4b4d7ece032880) and it seems to work. We were earlier always calling `Robolectric.flushBackgroundThreadScheduler()` twice which is why I had these redundant calls here but it doesn't seem necessary anymore. Thanks for pointing.